test/libsolidity/syntaxTests/inlineAssembly/prevrandao_reserved_paris.sol

```diff
@@ -0,0 +1,21 @@
+contract C {
+    function f() public view returns (uint256 ret) {
+        assembly {
+            let prevrandao := sload(0)
+            ret := prevrandao
+        }
+    }
+
+    function g() public pure returns (uint256 ret) {
+        assembly {
+            function prevrandao() -> r {
+                r := 1000
+            }
+            ret := prevrandao()
+        }
+    }
+}
+// ====
+// EVMVersion: >=paris
+// ----
+// ParserError 5568: (101-111): Cannot use builtin function name "prevrandao" as identifier name.
```

Comment:
This should probably only be a warning for now (independently of the EVM version) and then become an error (for any EVM version) with 0.9.

Reply:
Hum...I thought that the `prevrandao` would be disallowed as user-defined identifier since `prevrandao` is a builtin for `evm >= paris`. But I will emit the warning there as well.

Reply:
It would be better, if it *was* disallowed - but that's technically a breaking change. We'll switch to ``paris`` by default in the next release and then ``assembly { function prevrandao() {} }`` would suddenly become invalid in a minor version change... conversely, though, we don't want to wait for 0.9 with this. So keeping it allowed as identifier as well as a builtin, while emitting a warning for the use as identifier, is the compromise solution.
It won't happen in practice anyways, it's mainly to technically keep the change non-breaking.

With 0.9, we can properly reserve ``prevrandao()`` and disallow it as identifier in general, that'll clear things up again.